notebook/services/api/api.yaml

```diff
@@ -654,12 +639,18 @@ definitions:
       id:
         type: string
         format: uuid
-      notebook:
-        type: object
-        properties:
-          path:
-            type: string
-            description: path to notebook
+      path:
+        type: string
+        description: "The path to the session.  For console sessions, the path 
+         contains the cwd of the kernel and the name of the console session, 
+         which may be a uuid"
+      type:
+        type: string;
+        description: session type
+        enum:
+          - notebook
+          - file
+          - console
```

Comment:
A slight tangent:

I think the [EclairJS project](https://github.com/EclairJS/eclairjs-node) is using sessions in the kernel gateway to assign kernels a name and later look them up by that name. They do this to avoid having to track kernel IDs in their client and to rely on the already stateful kernel gateway to do it for them. 

They're currently using the `notebook` field in the current version of the API as the place to put that name. Path + type makes that feel a little less hacky, though path is still a bit of a misnomer (even the description above says it might contain a UUID) and none of the types named above really apply to the EclairJS use case. 

Is there a way to make these new field names more generically applicable and less specific to notebook / jupyter lab features? I realize this is opened against the notebook repo, but I'm thinking aloud about how it might cater to both these applications plus others in the more open ended use cases of the kernel gateway.